File: ingraph-antlr/src/main/java/cypher/main/MyCypherListener.java
package cypher.main;

import cypher.grammar.CypherBaseListener;
import cypher.grammar.CypherParser.MatchContext;
import cypher.grammar.CypherParser.ReturnContext;
import cypher.grammar.CypherParser.VariableContext;
import relalg.AlgebraExpression;

public class MyCypherListener extends CypherBaseListener {

	@Override
	public void enterMatch(MatchContext ctx) {
		System.out.println("Enter: " + ctx.getText());
		AlgebraExpression a;
	}

	@Override
	public void exitMatch(MatchContext ctx) {
		System.out.println("Exit: " + ctx.getText());
	}

	@Override
	public void enterVariable(VariableContext ctx) {
		System.out.println("var: " + ctx.getText());
	}

	@Override
	public void enterReturn(ReturnContext ctx) {
		System.out.println("Enter: " + ctx.getText());
	}

	@Override
	public void exitReturn(ReturnContext ctx) {
		System.out.println("Enter: " + ctx.getText());
	}
}

File: ingraph-antlr/src/main/kotlin/pkg/MyCypherListener.kt
package pkg

import cypher.grammar.CypherBaseListener
import cypher.grammar.CypherParser

class MyCypherListener : CypherBaseListenerListener {

    override fun enterMatch(ctx: CypherParser.MatchContext) {
        println("Enter: " + ctx.text)
    }

    override fun exitMatch(ctx: CypherParser.MatchContext) {
        println("Exit: " + ctx.text)
    }

    override fun enterVariable(ctx: CypherParser.VariableContext) {
        println("var: " + ctx.text)
    }

    override fun enterReturn(ctx: CypherParser.ReturnContext) {
        println("Enter: " + ctx.text)
    }

    override fun exitReturn(ctx: CypherParser.ReturnContext) {
        println("Enter: " + ctx.text)
    }
}
